pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +23,53 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, filename, relative_url=None):
+        """Upload a file to this repository.
+
+        The upload operation includes 4 steps:
+            1. Request an upload id from pulp
+            2. With the requested upload id, upload the file
+            3. After the upload's done, import the uploaded file to repository
+            4. Delete the upload request
+
+        Args:
+            filename (str)
+                Path to a local file to upload.
+
+            relative_url (str)
+                Path that should be used in remote repository
+
+                if omitted, filename will be used.
+        Returns:
+            Future[:class:`~pubtools.pulplib.Task`]
+                A future which is resolved when publish succeeds.
+
+                The future contains the task to delete upload request
+
+        Raises:
+            DetachedException
+                If this instance is not attached to a Pulp client.
+        """
+        if not self._client:
+            raise DetachedException()
+
+        path, name = os.path.split(filename.rstrip("/"))
```

Comment:
I don't understand the filename.rstrip ... if filename ends with "/" then it can't actually be a valid filename, can it?

For example, if I have a file "something.txt".  Why should I be able to upload_file("something.txt/") and have it automatically fixed to "something.txt"?  Shouldn't it just raise with an IO error since I gave an invalid filename?